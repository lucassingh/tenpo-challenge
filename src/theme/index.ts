import light from './lightTheme';

export default { light }