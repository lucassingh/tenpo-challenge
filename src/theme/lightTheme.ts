export default {
    colors: {
        primary: "#D4F9F5",
        secondary: "#00BAA4",
        background: "#F2F2F2",
        lightGray: "#F2F2F2",
        light: "#FFFFFF",
        dark: "#1C1C1C"
    }
}