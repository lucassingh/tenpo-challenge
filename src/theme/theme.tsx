import React from "react";
import { ThemeProvider } from "styled-components/native";
import theme from ".";

type Props = {
    children: React.ReactNode
}

export function Theme({children}: Props) {
    return <ThemeProvider theme={theme.light}>
        {children}
    </ThemeProvider>
}