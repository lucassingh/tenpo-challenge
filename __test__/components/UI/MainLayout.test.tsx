import React from "react";
import 'jest-styled-components'
import styled from 'styled-components/native';
import theme from "../../../src/theme";
import { render } from '@testing-library/react-native'

describe('Test main Layout component <Main />', () => {

     const Main = styled.View `  
        height: auto;
        flex: 1;
        background-color: ${theme.light.colors.background};
        border-top-right-radius: 35px;
        border-top-left-radius: 35px;
        margin-top: -55px;
        padding: 10px 0px 30px 15px;   
    `  
    
    it('Should match with snapshot', () => {

        const container  = render(<Main />);

        expect(container).toMatchSnapshot()
    })
})