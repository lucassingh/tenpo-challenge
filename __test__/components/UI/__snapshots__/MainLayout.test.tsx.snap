// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test main Layout component <Main /> Should match with snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F2F2F2",
        "borderTopLeftRadius": 35,
        "borderTopRightRadius": 35,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "auto",
        "marginTop": -55,
        "paddingBottom": 30,
        "paddingLeft": 15,
        "paddingRight": 0,
        "paddingTop": 10,
      },
    ]
  }
/>
`;
